region/wreckedship/main/Bowling Alley.json

```diff
@@ -274,6 +282,93 @@
         "h_canCrystalFlash"
       ]
     },
+    {
+      "link": [1, 1],
+      "name": "Bowling Alley Speed Block Moondance (Leave with Stored Fall Speed)",
+      "notable": true,
+      "requires": [
+        "SpeedBooster",
+        "h_canCrystalFlash",
+        "canInsaneJump",
+        "canTurnaroundAimCancel",
+        "canTemporaryBlue",
+        "canMoondance"
+      ],
+      "exitCondition": {
+        "leaveWithStoredFallSpeed": {
+          "fallSpeedInTiles": 1
+        }
+      },
+      "reusableRoomwideNotable": "Bowling Alley Speed Block Moondance",
+      "note": [
+        "Use SpeedBooster to construct a structure for Moondancing that has 2 top blocks intact, 2 middle blocks removed, and the bottom left block intact but the bottom right block removed.",
+        "Crystal Flash inside the middle hole to standup and then begin Moondancing.",
+        "After Samus sinks through the bottom tile, it is possible to wiggle right and then Turn-Around Aim Cancel to escape to the right."
+      ]
+    },
+    {
+      "link": [1, 1],
+      "name": "Bowling Alley Speed Block Moondance (Leave with More Stored Fall Speed)",
+      "notable": true,
+      "requires": [
+        "SpeedBooster",
+        "h_canCrystalFlash",
+        "canInsaneJump",
+        "canTurnaroundAimCancel",
+        "canTemporaryBlue",
+        "canExtendedMoondance",
+        "Grapple"
+      ],
+      "exitCondition": {
+        "leaveWithStoredFallSpeed": {
+          "fallSpeedInTiles": 2
+        }
+      },
+      "reusableRoomwideNotable": "Bowling Alley Speed Block Moondance",
+      "note": [
+        "Use SpeedBooster to construct a structure for Moondancing that has 2 top blocks intact, 2 middle blocks removed, and the bottom left block intact but the bottom right block removed.",
+        "Crystal Flash inside the middle hole to standup and then begin Moondancing.",
+        "Continue building fall speed by using Grappling Beam to return the the Moondance position.",
+        "Stop Moondancing exactly one Moonfall before Samus would sink through 2 tiles.",
+        "After Samus sinks through the bottom tile, it is possible to wiggle right and then Turn-Around Aim Cancel to escape to the right."
+      ]
+    },
+    {
+      "link": [1, 5],
```

Comment:
The possibility of getting into this area using a Crystal Flash as part of the moondance setup creates a risk of unsoundness in the logic in case a "Crystal Flash Grapple Clip" is needed to get out but without a way to damage down. The ability to damage down is probably something that we should eventually model in a systematic way, but that will take some time to figure out.

In the meantime, I think the problem could mostly be solved by adding an `{"obstaclesNotCleared": ["B"]}` requirement to the Crystal Flash Grapple Clip? There could be alternative requirements for cases where it would be possible to damage down within the room:
- If Phantoon is dead and it's possible to reach the Workrobots (i.e. with another Power Bomb), they could be used. Spikes could also be used, but since they deal 60 damage there isn't a guarantee you could get into the range from 1 to 50 (depending on the number of tanks collected). With a suit or a reserve tank, I guess spikes (if active) could always work.
- With SpeedBooster you could use shinesparks to damage down; this should maybe be a tech or helper, since it could be nullified by a randomizer "energy-free shinespark" option.

Reply:
I was distracted by the possibility of CFing in place of the xray clip that I forgot about that one.